dhcp-controller/controllers/dhcpserver_controller.go

```diff
@@ -173,114 +203,272 @@ func (r *DHCPServerReconciler) ensureServer(request reconcile.Request,
 	}
 
 	return nil, nil
 }
 
-// backendDeployment is a code for Creating Deployment
-func (r *DHCPServerReconciler) backendDeployment(v dhcpv1alpha1.DHCPServer) *appsv1.Deployment {
+// backendVM is a code for Creating VM
+func (r *DHCPServerReconciler) backendVM(v dhcpv1alpha1.DHCPServer) *unstructured.Unstructured {
 	networkNames, interfaceIps := parseNetwork(v.Spec.Networks)
-	size := int32(1)
-	memReq := resource.NewQuantity(64*1024*1024, resource.BinarySI)
-	memLimit := resource.NewQuantity(128*1024*1024, resource.BinarySI)
-	dockerRegistry := getRegistry(envVarDockerRegistry, defaultDockerRegistry)
-	image := dockerRegistry + "platform9/pf9-dnsmasq:v0.1"
-	log.Infof("image with registry %s", image)
-
-	dep := &appsv1.Deployment{
+	t := true
+	cores := resource.NewMilliQuantity(1000, resource.DecimalSI)
+	memReq := resource.NewQuantity(1024*1024*1024, resource.BinarySI)
+	memLimit := resource.NewQuantity(1024*1024*1024, resource.BinarySI)
+
+	// Make the cloudinit userData
+	tmpl, err := os.ReadFile("cloudinit.tmpl")
+	if err != nil {
+		log.Error(err, "unable to read template")
+	}
+	cloudinit := strings.Replace(string(tmpl), "{{.interfaceip}}", interfaceIps, 1)
+
+	cloudinit_secret := &corev1.Secret{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      v.Name,
+			Name:      v.Name + "-cloudinit",
+			Namespace: "default",
+		},
+		StringData: map[string]string{"userData": cloudinit},
+	}
+	err = r.Delete(context.TODO(), cloudinit_secret)
+	if err != nil {
+		log.Error(err, "Failed to delete the cloudinit secret or there is none")
+	}
+	err = r.Create(context.TODO(), cloudinit_secret)
+	if err != nil {
+		log.Error(err, "Failed to create the cloudinit secret")
+	}
+
+	// Make the cloudinit networkData
+	tmpl, err = os.ReadFile("netcloudinit.tmpl")
+	netcloudinit := "    version: 2\n    ethernets:\n"
+	for idx, interfaceIp := range strings.Split(interfaceIps, ",") {
+		str_tmpl := string(tmpl)
+		str_tmpl = strings.Replace(str_tmpl, "{{.num}}", strconv.Itoa(idx+1), 2)
+		netcloudinit = netcloudinit + strings.Replace(str_tmpl, "{{.ip}}", interfaceIp, 1)
+	}
+
+	netcloudinit_secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      v.Name + "-netcloudinit",
+			Namespace: "default",
+		},
+		StringData: map[string]string{"networkData": netcloudinit},
+	}
+	err = r.Delete(context.TODO(), netcloudinit_secret)
+	if err != nil {
+		log.Error(err, "Failed to delete the netcloudinit secret or there is none")
+	}
+	err = r.Create(context.TODO(), netcloudinit_secret)
+	if err != nil {
+		log.Error(err, "Failed to create the netcloudinit secret")
+	}
+
+	// Make the vm password secret
+	vmi_pwd_secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      v.Name + "-pwd",
 			Namespace: v.Namespace,
 		},
-		Spec: appsv1.DeploymentSpec{
-			Replicas: &size,
-			Selector: &metav1.LabelSelector{
-				MatchLabels: map[string]string{"app": "dnsmasq", "name": v.Name},
+		StringData: map[string]string{"root": "root"},
```

Comment:
We should be randomizing the password here. Ok to do in a separate change.

Reply:
noted